libportal/portal-private.h

```diff
@@ -71,3 +74,44 @@ const char * portal_get_bus_name (void);
 #define FLATPAK_PORTAL_BUS_NAME "org.freedesktop.portal.Flatpak"
 #define FLATPAK_PORTAL_OBJECT_PATH "/org/freedesktop/portal/Flatpak"
 #define FLATPAK_PORTAL_INTERFACE "org.freedesktop.portal.Flatpak"
+
+#if !GLIB_CHECK_VERSION(2, 76, 0)
+static inline gboolean
+g_clear_fd (int     *fd_ptr,
+            GError **error)
+{
+  int fd = *fd_ptr;
+
+  *fd_ptr = -1;
+
+  if (fd < 0)
+    return TRUE;
+
+  /* Suppress "Not available before" warning */
+  G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+  return g_close (fd, error);
+  G_GNUC_END_IGNORE_DEPRECATIONS
+}
+
+static inline void
+_g_clear_fd_ignore_error (int *fd_ptr)
+{
+  /* Don't overwrite thread-local errno if closing the fd fails */
+  int errsv = errno;
+
+  /* Suppress "Not available before" warning */
+  G_GNUC_BEGIN_IGNORE_DEPRECATIONS
+
+  if (!g_clear_fd (fd_ptr, NULL))
+    {
+      /* Do nothing: we ignore all errors, except for EBADF which
+       * is a programming error, checked for by g_close(). */
+    }
+
+  G_GNUC_END_IGNORE_DEPRECATIONS
+
+  errno = errsv;
+}
+
+#define g_autofd __attribute__((cleanup(_g_clear_fd_ignore_error)))
+#endif
```

Comment:
Hmmm I'd rather put this in a `glib-backports.h` file, and add an `#include "glib-backports.h` here

Reply:
done